### PR TITLE
Upgrade ndarray and other deps

### PR DESCRIPTION
**Pull request by Dirreke**
Bump ndarray from 0.15.6 to 0.16.0
Bump approx from 0.4.0 to 0.5
Bump criterion from 0.3.4 to 0.5.1

### CODE REVIEW

**Comment by Dirreke** [comment]
sorry, I forget to update tests file.  I will fix it later today

**Comment by Dirreke** [comment]
Fixed, Let's rerun CI

**Comment by bluss** [comment]
hi - I was here to press the ci button but I was wondering if @termoshtt wants to chime in too

**Comment by Dirreke** [comment]
It seems like we should bump MSRV. Or limit the version of clap. Which one would you like?
I could open a new PR to fix it.

**Comment by govinda-kamath** [comment]
Hi - any update on this. This seems to be holding back ndarray updates downstream?

File: ndarray-linalg/Cargo.toml [hunk 1]
```diff
@@ -37,7 +37,7 @@ rand = "0.8.3"
 thiserror = "1.0.24"
 
 [dependencies.ndarray]
-version = "0.15.2"
+version = "0.16.0"
```
Comment (govinda-kamath) [inline ndarray-linalg/Cargo.toml:40]:
```suggestion
version = ">=0.15.0, <0.17"
```

This would make it easier to use as a library downstream

**Comment by Garfield100** [comment]
@bluss @termoshtt @Dirreke 
Bumping this because this issue is quite painful while the fix is simple and a PR is already open and very close to being mergeable.
This crate hasn't received an update in 2 years and is currently unusable with other parts of the ecosystem. Don't let all the hard work which went into ndarray_linalg go to waste :)

**Comment by bluss** [comment]
I have been added as a crate owner for this crate on crates.io since today, thank to @termoshtt for graciously doing that.

I will help get the next release out as a transition thing - but those that need this package are hereby invited to become maintainers.

**Pull request closed without merging**